src/nvim/runtime.c

```diff
@@ -269,9 +269,25 @@ void set_context_in_runtime_cmd(expand_T *xp, const char *arg)
   xp->xp_pattern = (char *)arg;
 }
 
-static void source_callback(char *fname, void *cookie)
+static void source_callback(int num_fnames, char **fnames, void *cookie)
 {
-  (void)do_source(fname, false, DOSO_NONE, cookie);
+  for (int i = 0; i < num_fnames; i++) {
+    if (str_ends_with(fnames[i], ".vim")) {
+      (void)do_source(fnames[i], false, DOSO_NONE, cookie);
+    }
+  }
+
+  for (int i = 0; i < num_fnames; i++) {
+    if (str_ends_with(fnames[i], ".lua")) {
+      (void)do_source(fnames[i], false, DOSO_NONE, cookie);
+    }
+  }
+
+  for (int i = 0; i < num_fnames; i++) {
+    if (!str_ends_with(fnames[i], ".vim") && !str_ends_with(fnames[i], ".lua")) {
+      (void)do_source(fnames[i], false, DOSO_NONE, cookie);
```

Comment:
Is it not worth the code simplification to combine all these into the same for-loop, and would that preserve the "old" behavior?